src/ert/storage/local_ensemble.py

```diff
@@ -670,10 +727,30 @@ def save_field(
         parameter_name: str,
         realization: int,
         data: npt.ArrayLike,
+        unmasked: bool = False,
     ) -> None:
         output_path = self.mount_point / f"realization-{realization}"
         Path.mkdir(output_path, exist_ok=True)
-        np.save(f"{output_path}/{parameter_name}", data)
+
+        if unmasked:
+            grid = self.experiment.grid
+            assert grid
+            if isinstance(grid, xtgeo.Grid):
+                masked_data = [np.nan] * grid.ntotal
+                for i, e in enumerate(grid.actnum_indices):
```

Comment:
Have you perhaps tried writing this without the for-loop and instead relying on numpy-indexing?
Something like:

```python
masked_data[grid.actnum_indices] = data[grid.actnum_indices]
```

Fields can have tens of millions of elements so it might be worth a try.

Reply:
Agree